src/roles/harlot.py

```diff
@@ -140,6 +134,15 @@ def on_transition_night_end(evt, var):
 @event_listener("begin_day")
 def on_begin_day(evt, var):
     VISITED.clear()
+    PASSED.clear()
+
+@event_listener("swap_player")
+def on_swap(evt, var, old_user, user):
+    for actor, target in set(VISITED.items()):
+        if target is old_user:
+            VISITED[actor] = user
+        if actor is old_user:
+            VISITED[user] = VISITED.pop(actor)
```

Comment:
You aren't fixing PASSED in swap_player. I'd also prefer renaming "actor" to "harlot" for extra descriptiveness :)